### PR TITLE
[RHSSO-1431] Introduce image stream and application templates for the RH-SSO 7.3 TP CD image

### PR DESCRIPTION
**Pull request by iankko**
Set the ```version``` and ```xpaas``` template fields to ```7.3.0.CD02``` value

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by iankko** [comment]
@rcernich @Crumby @maschmid PTAL (once got a chance)

Have tested the change, and the templates seem to be working fine (the 7.3.0.CD02 server is up and running once deployed):

pod_log

and

server_info

But I am not sure, about the following two things:
- If it's OK to replace the ```1.4.14``` version with ```1.0``` one?, and
- Unsure, which value to use for the ```xpaas``` label (for now / temporarily used the ```1.5.0``` version, like the EAP CD template do). But should this be set to something else? Or even the label removed, and replaced with something else? Advice / suggestions welcome.

Thanks, Jan

**Comment by rcernich** [comment]
I think for EAP CD, the image tag/version matches the CD version, so 2.0 in this case.

**Comment by rcernich** [comment]
Per my last comment, do you want the version to be CD 2 or 7.3 CD 2?  In the EAP case, it's just EAP CD 12, for example.

**Comment by iankko** [comment]
@rcernich Thanks for suggestions, Rob!

The ```7.3.0.CD02``` value makes sense to me, since the CD version is the milestone we might be potentially needing to update the templates for. Set both the ```version``` and ```xpaas``` fields to this value. Used ```7.3.0``` in the version too, for the case, there would be ```7.4.0.CD02``` one in the future, so we wouldn't possibly conflict.

**Comment by iankko** [comment]
@goldmann jFYI. Maybe the sanity of this one could be tested via modified / updated ARQ tests?

**Comment by Crumby** [comment]
@iankko templates and image stream passed tests...

**Comment by iankko** [comment]
@Crumby Woot, great news, Pavel! Thank you. So any objections against merging this?

@rcernich You OK with the form of ```version``` and ```xpaas``` label fields in the current version?

File: templates/sso-cd-image-stream.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                "name": "redhat-sso-cd-openshift",
+                "annotations": {
+                    "description": "Red Hat Single Sign-On 7.3 continuous delivery",
+                    "openshift.io/display-name": "Red Hat Single Sign-On 7.3 continuous delivery",
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "version": "7.3.0.CD02"
+                }
+            },
+            "labels": {
+                "xpaas": "7.3.0.CD02"
```
Comment (rcernich) [inline templates/sso-cd-image-stream.json]: I'd replace xpaas with rhsso.

File: templates/sso-cd-mysql-persistent.json [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+            "description": "An example application based on RH-SSO 7.3 CD Tech Preview image. For more information about using this template, see https://github.com/jboss-container-images/redhat-sso-7-openshift-image/docs.",
+            "template.openshift.io/long-description": "This template defines resources needed to develop Red Hat Single Sign-On 7.3 server based deployment, deployment configuration for MySQL using persistence, and securing RH-SSO communication using passthrough TLS.",
+            "template.openshift.io/documentation-url": "https://access.redhat.com/documentation/en/red-hat-single-sign-on/",
+            "template.openshift.io/support-url": "https://access.redhat.com"
+        },
+        "name": "sso-cd-mysql-persistent"
+    },
+    "labels": {
+        "template": "sso-cd-mysql-persistent",
+        "xpaas": "7.3.0.CD02"
```
Comment (rcernich) [inline templates/sso-cd-mysql-persistent.json]: same here, rhsso instead of xpaas.

Comment (iankko) [reply templates/sso-cd-mysql-persistent.json]: Ok, thanks! I will do sed global replace for them.

**Comment by iankko** [comment]
Replaced all occurrences.

**Comment by iankko** [comment]
Merging based on approval from Pavel Drobek (GH nickname @Crumby) above. Thanks, Pavel!

**Pull request merged**